aten/src/ATen/native/cuda/Activation.cu

```diff
@@ -24,6 +24,93 @@
 namespace at {
 namespace native {
 
+// -----------------------------------
+// glu forward
+// -----------------------------------
+void glu_kernel(TensorIterator& iter) {
+  AT_DISPATCH_FLOATING_TYPES_AND2(kHalf, kBFloat16, iter.dtype(), "glu_cuda", [&]() {
+    using acc_t = at::acc_type<scalar_t, true>;
+    gpu_kernel(iter, [] GPU_LAMBDA (scalar_t a_, scalar_t b_) -> scalar_t {
+      const acc_t a = a_;
+      const acc_t b = b_;
+      const acc_t one = acc_t(1);
+      const acc_t sigmoid = one / (one + std::exp(-b));
+      return a * sigmoid;
+    });
+  });
+}
+
+// -----------------------------------
+// glu backward
+// -----------------------------------
+template <typename scalar_t, typename OffsetCalc, typename StrideType>
+__global__ void glu_backward_kernel(
```

Comment:
why do you need a one-off kernel for this instead of TensorIterator? The reason it was done this way in THC was that the number of inputs was limited to 3, so these tricks were needed, but with TI there can be arbitrary many inputs, and thus backward can be a regular TI kernel.

Reply:
I tried with `gpu_kernel_multiple_outputs` and it's ~1.5x slower in cuda time and also has greater CPU overhead since we need dispatch 4 times to narrow the inputs/outputs.